standard/structs.md

```diff
@@ -476,3 +501,71 @@ Static constructors for structs follow most of the same rules as for classes. Th
+
+- For a return statement `return e1`, the safe-scope of `e1` must be calling-method.
+- For an assignment `e1 = e2` the safe-scope of `e2` must be at least as wide a scope as the safe-scope of `e1`.
+
+For a method invocation if there is a `ref` or `out` argument of a `ref struct` type (including the receiver unless the type is `readonly`), with safe-scope `E1`, then no argument (including the receiver) may have a narrower safe-scope than `E1`.
+
+Any expression whose compile-time type is not a ref struct has a safe-scope of calling-method.
+
+A `default` expression, for any type, has safe-scope of calling-method.
+
+#### safe-scope-rules-parameter Parameter safe scope
+
+A formal parameter of a ref struct type, including the `this` parameter of an instance method, has a safe-scope of calling-method.
+
+#### safe-scope-rules-local Local variable safe scope
+
+A local variable of a ref struct type has a safe-scope as follows:
+
+- If the variable is an iteration variable of a `foreach` loop, then the variable's safe-scope is the same as the safe-scope of the `foreach` loop's expression.
+- A local of `ref struct` type uninitialized at the point of declaration has a safe-scope of calling method.
+- Otherwise the variable's declaration requires an initializer. The variable's safe-scope is the same as the safe-scope of its initializer.
+
+#### safe-scope-rules-field Field safe scope
+
+A a reference to a field, `e.F` where `F` is a ref struct type, has a safe-scope that is the same as the safe-scope of `e`.
+
+#### safe-scope-rules-operator Operators
+
+The application of a user-defined operator is treated as a method invocation (§safe-scope-method-invocation).
+
+For an operator that yields a value, such as `e1 + e2` or `c ? e1 : e2`, the safe-scope of the result is the narrowest scope among the safe-scopes of the operands of the operator. As a consequence, for a unary operator that yields a value, such as `+e`, the safe-scope of the result is the safe-scope of the operand.
+
+#### §safe-scope-method-invocation Method and property invocation
+
+A value resulting from a method invocation `e1.M(e2, ...)` has safe-scope of the smallest of the following scopes:
+
+- calling-method.
+- The safe-scope of all argument expressions (including the receiver).
+
+A property invocation (either `get` or `set`) it treated as a method invocation of the underlying method by the above rules.
+
+#### safe-scope-rules-stackalloc `stackalloc`
+
+The result of a stackalloc expression has safe-scope of current-method. It is not safe-to-return from the method.
+
+#### safe-scope-rules-constructor Constructor invocations
+
+A `new` expression that invokes a constructor obeys the same rules as a method invocation that is considered to return the type being constructed.
+
+In addition the safe-scope is no wider than the smallest of the safe-scopes of all arguments and operands of all object initializer expressions, recursively, if any initializer is present.
```

Comment:
Can we be more specific than "no wider", e.g. just "is the smallest of the safe-scopes [...]"?